source/auth/tests/README.rst

```diff
@@ -0,0 +1,55 @@
+==========
+Auth Tests
+==========
+
+The YAML and JSON files in this directory tree are platform-independent tests
+that drivers can use to prove their conformance to the Auth Spec at least with
+respect to connection string URI input.
+
+Drivers should do additional unit testing if there are alternate ways of
+configuring credentials on a client.
+
+Driver must also conduct the prose tests in the Auth Spec test plan section.
+
+Format
+------
+
+Each YAML file contains an object with a single ``tests`` key. This key is an
+array of test case objects, each of which have the following keys:
+
+- ``description``: A string describing the test.
+- ``uri``: A string containing the URI to be parsed.
+- ``valid:`` A boolean indicating if the URI should be considered valid.
+- ``credential_configured:`` A boolean indicating if the URI is considered to have
```

Comment:
I had wondered about this given the previous format, but is there a reason we can't just set `auth` (or `credential` now) to `null` in the event that no credential should be generated? I'm not sure I see why we need another field?

Reply:
Jeff seems to want to use null for "should not exist", so if we make that change for elements in credential, we can do it at the top level, too.

Reply:
This is due to legacy decisions to have `null` imply "ignore this assertion". An alternative might be to amend the grammar to allow `credential` to be one of `null` (ignore), boolean `false`, or a document.